modules/barcode/src/detector/bardetect.cpp

```diff
@@ -0,0 +1,422 @@
+        width = cvRound(src.size().width / coeff_expansion);
+        height = cvRound(src.size().height / coeff_expansion);
+        Size new_size(width, height);
+        resize(src, resized_barcode, new_size, 0, 0, INTER_AREA);
+    }
+//    else if (min_side < 512.0)
+//    {
+//        purpose = ZOOMING;
+//        coeff_expansion = 512.0 / min_side;
+//        width = cvRound(src.size().width * coeff_expansion);
+//        height = cvRound(src.size().height * coeff_expansion);
+//        Size new_size(width, height);
+//        resize(src, resized_barcode, new_size, 0, 0, INTER_CUBIC);
+//    }
+    else
+    {
+        purpose = UNCHANGED;
+        coeff_expansion = 1.0;
+        width = src.size().width;
+        height = src.size().height;
+        resized_barcode = src.clone();
+    }
+//    medianBlur(resized_barcode, resized_barcode, 3);
+
+}
+
+
+void Detect::localization()
+{
+
+    localization_bbox.clear();
+    bbox_scores.clear();
+
+    // get integral image
+    preprocess();
+    // empirical setting
+    static constexpr float SCALE_LIST[] = {0.01f, 0.03f, 0.06f, 0.08f};
+    const auto min_side = static_cast<float>(std::min(width, height));
+    int window_size;
+    for (const float scale:SCALE_LIST)
+    {
+        window_size = cvRound(min_side * scale);
+        calCoherence(window_size);
+        barcodeErode();
+        regionGrowing(window_size);
+    }
+
+}
+
+
+bool Detect::computeTransformationPoints()
+{
+
+    bbox_indices.clear();
+    transformation_points.clear();
+    transformation_points.reserve(bbox_indices.size());
+    RotatedRect rect;
+    Point2f temp[4];
+    const float THRESHOLD_SCORE = float(width * height) / 500.f;
+    dnn::NMSBoxes(localization_bbox, bbox_scores, THRESHOLD_SCORE, 0.1f, bbox_indices);
+
+    for (const auto &bbox_index : bbox_indices)
+    {
+        rect = localization_bbox[bbox_index];
+        if (purpose == ZOOMING)
+        {
+            rect.center /= coeff_expansion;
+            rect.size.height /= static_cast<float>(coeff_expansion);
+            rect.size.width /= static_cast<float>(coeff_expansion);
+        }
+        else if (purpose == SHRINKING)
+        {
+            rect.center *= coeff_expansion;
+            rect.size.height *= static_cast<float>(coeff_expansion);
+            rect.size.width *= static_cast<float>(coeff_expansion);
+        }
+        rect.points(temp);
+        transformation_points.emplace_back(vector<Point2f>{temp[0], temp[1], temp[2], temp[3]});
+    }
+
+    return !transformation_points.empty();
+}
+
+
+void Detect::preprocess()
+{
+    Mat scharr_x(resized_barcode.size(), CV_32F), scharr_y(resized_barcode.size(), CV_32F), temp;
+    Scharr(resized_barcode, scharr_x, CV_32F, 1, 0);
+    Scharr(resized_barcode, scharr_y, CV_32F, 0, 1);
+    // calculate magnitude of gradient, normalize and threshold
+    magnitude(scharr_x, scharr_y, gradient_magnitude);
+    threshold(gradient_magnitude, gradient_magnitude, 56, 1, THRESH_BINARY);
```

Comment:
> 56

Please avoid using of magic numbers.
Use dedicated const properly named variable for that.